### PR TITLE
feat: Add Achievements packages

### PR DESCRIPTION
**Pull request by ChefMomota**
---

## PR-Codex overview
The focus of this PR is to refactor and enhance the achievements module by restructuring code, adding new types, and updating dependencies.

### Detailed summary
- Refactored achievements module structure
- Added new types for campaigns
- Updated dependencies and configurations
- Replaced imports with achievements module references

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by chef-jojo** [comment]
Why do we need to move to pkg? Will it be used by other apps?

**Comment by ChefMomota** [comment]
> Why do we need to move to pkg? Will it be used by other apps?

Yes. We need to use it in apps/gamification profile display.

File: packages/achievements/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "version": "0.0.1",
+  "sideEffects": false,
+  "private": true,
+  "scripts": {
+    "test": "vitest --run"
+  },
+  "type": "module",
+  "dependencies": {
+    "qs": "^6.0.0",
+    "lodash": "^4.17.21"
```
Comment (chefjackson) [inline packages/achievements/package.json:12]: Do we need these two dependencies?

Comment (ChefMomota) [reply packages/achievements/package.json]: done

**Review by chefjackson — APPROVED** [review]

**Pull request merged**